homeassistant/components/roborock/roborock_storage.py

```diff
@@ -0,0 +1,129 @@
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        entry_id: str,
+    ) -> None:
+        """Initialize RoborockStorage."""
+        self._hass = hass
+        self._data: dict[str, RoborockMapEntry] = {}
+        self._entry_id = entry_id
+
+    def _should_update(self, map_entry: RoborockMapEntry | None):
+        return (
+            map_entry is None
+            or dt_util.utcnow().timestamp() - map_entry.time > MAP_UPDATE_FREQUENCY
+        )
+
+    def _get_map_filename(self, map_name: str):
+        return self._hass.config.path(f"{MAP_PATH}/{self._entry_id}/{map_name}")
+
+    def exec_load_maps(self, map_names: list[str]) -> list[bytes | None]:
+        """Load map content. Should be called in executor thread."""
+        filenames: list[tuple[str, str]] = [
+            (map_name, self._get_map_filename(map_name)) for map_name in map_names
+        ]
+
+        results: list[bytes | None] = []
+        for map_name, filename in filenames:
+            try:
+                if not os.path.exists(filename):
+                    map_data = None
+                else:
+                    _LOGGER.debug("Reading map from disk store: %s", filename)
+                    with open(filename, "rb") as stored_map:
+                        map_data = stored_map.read()
+            except OSError as err:
+                _LOGGER.error("Unable to read map file: %s %s", filename, err)
+                results.append(None)
+                continue
+            if map_data is None:
+                results.append(None)
+                continue
+            self._data[map_name] = RoborockMapEntry(
+                map_name,
+                dt_util.utcnow().timestamp(),
+            )
+            results.append(map_data)
+        return results
+
+    def _save_map(self, filename: str, content: bytes) -> None:
+        """Help other functions save the map. Should not be called separately."""
+        os.makedirs(os.path.dirname(filename), exist_ok=True)
+        _LOGGER.debug("Saving event map to disk store: %s", filename)
+        with open(filename, "wb") as stored_map:
+            stored_map.write(content)
+
+    async def async_save_map(self, map_name: str, content: bytes) -> None:
+        """Write map if it should be updated."""
+        map_entry = self._data.get(map_name)
+        if not self._should_update(map_entry):
+            return None
+        filename = self._get_map_filename(map_name)
+        self._data[map_name] = RoborockMapEntry(map_name, dt_util.utcnow().timestamp())
+
+        try:
+            await self._hass.async_add_executor_job(self._save_map, filename, content)
+        except OSError as err:
+            _LOGGER.error("Unable to write map file: %s %s", filename, err)
+            # We don't want the _data dict to be updated with incorrect information
+            # Revert it if it previously existed.
+            if map_entry is not None:
+                self._data[map_name] = map_entry
```

Comment:
This function body can be replaced with
```suggestion
        await self.async_save_maps([(map_name, content)])
```

Or maybe we remove usage of this function completely?